### PR TITLE
Fix/3215 toolbar long text in toolbar shows overflow menu prematurely with scrollbar

### LINKED ISSUE #3167: [Toolbar] Action bar shows scrollbar
### Provide a general summary of the issue here

a scrollbar is visible in the action bar if there are a lot of controls

Image

### 🤔 Expected Behavior?

scrollbars should not be visible

### 😯 Current Behavior

-

### 💁 Possible Solution

-

### 🖥️ Steps to Reproduce

Use the story below to reproduce this behavior

```typescript
export const Multiple: Story = {
  args: {
    selectAll: false
  },
  render: ({ selectAll }, { loaded: { students } }) => {
    return html`
      <sl-grid .items=${(students as Student[]).slice(0, 5)}>
        <sl-grid-selection-column ?select-all=${selectAll}></sl-grid-selection-column>
        <sl-grid-column
          header="Student"
          path="fullName"
          .renderer=${avatarRenderer}
          .scopedElements=${{ 'sl-avatar': Avatar }}
        ></sl-grid-column>
        <sl-grid-column path="email"></sl-grid-column>

        <!-- These get slotted into the floating tool-bar -->

        <sl-tooltip id="tooltip-edit"> Collective Change </sl-tooltip>
        <sl-tooltip id="tooltip-delete"> Delete </sl-tooltip>

        <sl-button
          id="tooltip-edit"
          fill="outline"
          slot="bulk-actions"
          variant="inverted"
          aria-label="Duplicate"
          aria-describedby="tooltip-edit"
        >
          <sl-icon name="far-copy"></sl-icon>
        </sl-button>

        <sl-button id="tooltip-edit" fill="outline" slot="bulk-actions" variant="inverted" aria-label="Duplicate">
          <sl-icon name="far-copy"></sl-icon>
        </sl-button>

        <sl-menu-button variant="inverted" fill="outline" aria-label="Visibility" slot="bulk-actions">
          <sl-icon name="far-right-to-line" slot="button"></sl-icon>
          <span slot="button">Visibility</span>
          <sl-menu-item>Hide selection</sl-menu-item>
          <sl-menu-item>Show selection</sl-menu-item>
        </sl-menu-button>

        <sl-menu-button variant="inverted" fill="outline" aria-label="Block" slot="bulk-actions">
          <sl-icon name="far-right-to-line" slot="button"></sl-icon>
          <span slot="button">Block</span>
          <sl-menu-item>Block selection</sl-menu-item>
          <sl-menu-item>Unblock selection</sl-menu-item>
        </sl-menu-button>

        <sl-button fill="outline" slot="bulk-actions" variant="inverted">
          <sl-icon name="far-trash"></sl-icon>
          Create calculated calculated column
        </sl-button>
      </sl-grid>
    `;
  }
};
```

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### If other, please specify.

_No response_

### What operating system are you using?

macos

### 👤 Your name

Frans de Boer

### 🧢 Your product/team

Magister

### 🎨 Your Theme(s)

Magister

### LINKED ISSUE #3215: [Toolbar] long text in toolbar shows overflow menu prematurely with scrollbar
### Provide a general summary of the issue here

a long button text shows the overflow state even if there is enough room and a scrollbar is displayed

this issue is introduced since grid version 0.7.1.

Image

in this screenshot the overflow state is active
Image

### 🤔 Expected Behavior?

should only overflow if there is not enough room

### 😯 Current Behavior

-

### 💁 Possible Solution

-

### 🖥️ Steps to Reproduce

Please use the Story below. Select some rows and resize the browser window.

```typescript
export const Multiple: Story = {
  args: {
    selectAll: false
  },
  render: ({ selectAll }, { loaded: { students } }) => {
    return html`
      <sl-grid .items=${(students as Student[]).slice(0, 5)}>
        <sl-grid-selection-column ?select-all=${selectAll}></sl-grid-selection-column>
        <sl-grid-column
          header="Student"
          path="fullName"
          .renderer=${avatarRenderer}
          .scopedElements=${{ 'sl-avatar': Avatar }}
        ></sl-grid-column>
        <sl-grid-column path="email"></sl-grid-column>

        <sl-button fill="outline" slot="bulk-actions" variant="inverted">
          <sl-icon name="far-copy"></sl-icon>
          Duplicate
        </sl-button>
        <sl-button fill="outline" slot="bulk-actions" variant="inverted">
          <sl-icon name="far-trash"></sl-icon>
          Delete
        </sl-button>
        <sl-button disabled fill="outline" slot="bulk-actions" variant="inverted">
          <sl-icon name="far-right-to-line"></sl-icon>
          Action 1
        </sl-button>

        <sl-menu-button variant="inverted" fill="outline" aria-label="Block" slot="bulk-actions">
          <sl-icon name="far-eye-slash" slot="button"></sl-icon>
          <span slot="button"> ${msg('Visibility')} </span>
          <sl-menu-item>Block</sl-menu-item>
          <sl-menu-item>Unblock</sl-menu-item>
        </sl-menu-button>

        <sl-menu-button variant="inverted" fill="outline" aria-label="Visibility" slot="bulk-actions">
          <sl-icon name="far-eye-slash" slot="button"></sl-icon>
          <span slot="button"> ${msg('Visibility')} </span>
          <sl-menu-item>Hide Selection</sl-menu-item>
          <sl-menu-item>Show Selection</sl-menu-item>
        </sl-menu-button>

        <sl-button ${tooltip('I am a tooltip')} fill="outline" slot="bulk-actions" variant="inverted">
          <sl-icon name="far-right-to-line"></sl-icon>
          Nieuwe berekeningskolom aanmaken
        </sl-button>
        <sl-button fill="outline" slot="bulk-actions" variant="inverted">
          <sl-icon name="far-right-to-line"></sl-icon>
          Action 3
        </sl-button>
      </sl-grid>
    `;
  }
};
```

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### If other, please specify.

_No response_

### What operating system are you using?

macos

### 👤 Your name

Frans de Boer

### 🧢 Your product/team

Magister

### 🎨 Your Theme(s)

Magister

### PR DESCRIPTION
**Pull request by anna-lach**
Fixes:
- #3215
- #3167

This pull request introduces several improvements and fixes to the tool-bar and grid components, focusing on more accurate measurement and handling of overflow, as well as UI enhancements for bulk actions and layout containment. The most significant changes are in the tool-bar's overflow logic, measurement utilities, and storybook examples for the grid.

**Tool-bar measurement and overflow handling:**

* Refactored overflow logic in `overflow.ts` to more accurately calculate visible items, reserve space for the menu button, and ensure that when all items are hidden, the menu button's margin is handled correctly. Introduced a helper function `setItemVisibility` for clarity. [[1]](diffhunk://#diff-87f19e4397f72b4b0cff3bdcb16e11909e94027993f5e6ca4b447f03aa074c18L40-R59) [[2]](diffhunk://#diff-87f19e4397f72b4b0cff3bdcb16e11909e94027993f5e6ca4b447f03aa074c18R79-R98)
* Added new utility functions: `measureMenuButtonWidth` (measures menu button width including margin, with fallbacks) and `measureConstrainedWidth` (measures available width with containment, preventing parent growth and layout issues).
* Updated `tool-bar.ts` to use the new measurement utilities, improving how available width and menu button width are calculated for overflow logic. [[1]](diffhunk://#diff-577451be1be41a4f289ae24ec8d8a8f742df1254fb80a04465d842ffc66befaaL13-R20) [[2]](diffhunk://#diff-577451be1be41a4f289ae24ec8d8a8f742df1254fb80a04465d842ffc66befaaL366-R383)
* Improved measurement reliability in `#measureItems()` by skipping recalculation only when the toolbar is hidden, not when there is no overflow.
* Added `min-inline-size: 0` and a `:host(:state(measuring))` rule with `contain: inline-size` to `tool-bar.scss`, supporting more accurate width measurements and preventing layout shifts.

**Grid component UI improvements:**

* Added two long action buttons to the bulk actions slot in the grid selection story, demonstrating how the UI handles overflow and long button labels.
* Added `overflow: clip` to the grid's table footer for improved layout and to prevent unwanted scrollbars or overflows.

### CODE REVIEW

File: packages/components/tool-bar/src/overflow.spec.ts [hunk 1]
```diff
@@ -144,101 +144,138 @@ describe('calculateVisibility', () => {
 });
 
 describe('overflow (integration)', () => {
```
Comment (michal-sanoma) [inline packages/components/tool-bar/src/overflow.spec.ts:144]: **suggestion:** The toolbar tests are solid, but this bug was reported in the grid bulk-actions context. Could we add a regression test that specifically covers long bulk-action labels in grid and verifies we don’t show overflow and don’t get a scrollbar when there is still enough room?

**Comment by Diaan** [comment]
This scenario is broken: https://jolly-stone-0f7fffc03-3241.westeurope.6.azurestaticapps.net/?path=/story/actions-tool-bar--fit-content

As well as these: https://jolly-stone-0f7fffc03-3241.westeurope.6.azurestaticapps.net/?path=/story/actions-tool-bar--examples
(the last button and the menu button flicker)

**Comment by Diaan** [comment]
Another bug: https://jolly-stone-0f7fffc03-3241.westeurope.6.azurestaticapps.net/?path=/story/grid-selection--multiple

If you make the screen smaller and then bigger again it starts flickering:

https://github.com/user-attachments/assets/b23ae671-4f2f-401d-935a-02972f7c95f8

(only happens when the last button comes into view

**Review by michal-sanoma — APPROVED** [review]
Great work! 🚀
